internal/controller/bucket/bucket.go

```diff
@@ -139,42 +138,50 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	// observed only on this S3 Backend. An empty config reference name will be automatically set
 	// to "default".
 	if cr.GetProviderConfigReference() != nil && cr.GetProviderConfigReference().Name != defaultPC {
-		bucketExists, err := c.bucketExists(ctx, cr.GetProviderConfigReference().Name, cr.Name)
-		if err != nil {
-			return managed.ExternalObservation{}, err
-		}
-		if bucketExists {
-			return managed.ExternalObservation{
-				// Return false when the external resource does not exist. This lets
-				// the managed resource reconciler know that it needs to call Create to
-				// (re)create the resource, or that it has successfully been deleted.
-				ResourceExists: true,
+		return c.observe(ctx, cr)
```

Comment:
In my perspective observe and observeAll do the same, I mean a function which is able to do a thing on multiple items is able to do the same on a single item. Why we need two functions?

Reply:
Good point. There's no need to limit the observation to a single backend :+1: